### PR TITLE
Feature/save recipes

### PR DESCRIPTION
**Pull request by LordSchwifty**
Files Changed: index.html, scripts.js, images, styles.css
Change Summary: Changed the saved recipe icon and gave it functionality by checking if a recipe is saved or not and toggling the hidden class. Set up a framework to connect the dom and user classes with saving recipes. Added event listeners to save button and refactored recipe modal functionality.

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -37,32 +41,40 @@ function createRecipeCards(recipes) {
...
     });
 }
 
 function createRecipeModal(event) {
   toggleHidden(modalSection);
   let recipeID = +(event.target.dataset.parent);
-  let clickedRecipe = recipeRepo.recipes.find(recipe => recipe.id === recipeID);
+  modalRecipe = recipeRepo.recipes.find(recipe => recipe.id === recipeID);
+  let buttonText
+  modalRecipe.saved ? buttonText = "Remove from Saved Recipes" : buttonText = "Add to Saved Recipes"
```
Comment (qrispi) [inline src/scripts.js:59]: Great use of a ternary here!

File: src/scripts.js [hunk 2]
```diff
@@ -15,12 +15,16 @@ const recipeSection = document.getElementById('allRecipes');
 const modalSection = document.getElementById('recipeModalBackground');
 const filterDropdown = document.getElementById('filterDropdown');
 const searchBar = document.getElementById('searchBar')
+let modalRecipe 
+// const saveButton = document.getElementById('saveBtn')
```
Comment (qrispi) [inline src/scripts.js:19]: Love the global modalRecipe. Let's remember to delete this unused line of code below it too!

**Review by qrispi — APPROVED** [review]
Loving the progress! We are almost there!

File: src/scripts.js [hunk 3]
```diff
@@ -37,32 +41,40 @@ function createRecipeCards(recipes) {
...
   toggleHidden(modalSection);
   let recipeID = +(event.target.dataset.parent);
-  let clickedRecipe = recipeRepo.recipes.find(recipe => recipe.id === recipeID);
+  modalRecipe = recipeRepo.recipes.find(recipe => recipe.id === recipeID);
+  let buttonText
+  modalRecipe.saved ? buttonText = "Remove from Saved Recipes" : buttonText = "Add to Saved Recipes"
   modalSection.innerHTML = `
   <div class="recipe-popup">
-      <h2>${clickedRecipe.name}</h2>
+      <h2>${modalRecipe.name}</h2>
```
Comment (jfogiato) [inline src/scripts.js:62]: Definitely agree with the variable name change here - good call!

File: src/scripts.js [hunk 4]
```diff
@@ -37,32 +41,40 @@ function createRecipeCards(recipes) {
...
       <ol class="direction-list">
       <h3>Directions:</h3>
-      ${createList(clickedRecipe.getInstructions())}
+      ${createList(modalRecipe.getInstructions())}
       </ol>
-      <h4>TOTAL COST $${+(clickedRecipe.listCost(ingredientsData))}</h4>
+      <h4>TOTAL COST $${+(modalRecipe.listCost(ingredientsData))}</h4>
+      <button class="save-button" id="saveBtn">${buttonText}</button>
   </div>`;
+  document.getElementById('saveBtn').addEventListener('click', toggleSaveRecipe);
```
Comment (jfogiato) [inline src/scripts.js:77]: May want to revisit this later. Even though we're regenerating the HTML for this section with each popup modal, is it possible that we would still have eventListeners out there lingering? If so, we may want to change the element we're listening on to the modal (or add a conditional to check if the event is happening on the button?) and then change our handler to account for that.

File: src/scripts.js [hunk 5]
```diff
@@ -92,4 +104,9 @@ function searchRecipes() {
   let keyword = searchBar.value;
   let searchedRecipes = recipeRepo.filterByName(keyword);
   createRecipeCards(searchedRecipes);
+}
+
+function toggleSaveRecipe(event) {
```
Comment (jfogiato) [inline src/scripts.js:109]: Assuming this is just a stand-in to check functionality?

File: src/styles.css [hunk 6]
```diff
@@ -48,6 +49,14 @@ h3 {
   width: 35vmin;
 }
 
+.heart-icon {
```
Comment (jfogiato) [inline src/styles.css:52]: RIP ⭐

**Review by jsahim — APPROVED** [review]

**Pull request merged**